mobile/library/common/extensions/filters/http/network_configuration/filter.cc

```diff
@@ -105,23 +107,17 @@ NetworkConfigurationFilter::resolveProxy(Http::RequestHeaderMap& request_headers
   Network::ProxyResolutionResult proxy_resolution_result = proxy_resolver->resolver->resolveProxy(
       target_url, proxy_settings_,
       [&weak_self](const std::vector<Network::ProxySettings>& proxies) {
+        ASSERT(Thread::MainThread::isMainOrTestThread());
```

Comment:
I wonder if this should be a release assert since it is otherwise a no-op on release builds?

Reply:
Good point, especially since we can turn off the experiment that runs this code, a release assert would better tell us if we actually have a threading issue or not.  Addressed in https://github.com/envoyproxy/envoy/pull/35155